sdk/program/src/message/sanitized.rs

```diff
@@ -192,10 +189,13 @@ impl SanitizedMessage {
 
     /// Returns true if the account at the specified index is writable by the
     /// instructions in this message.
-    pub fn is_writable(&self, index: usize) -> bool {
+    pub fn is_writable(&self, index: usize, demote_program_write_locks: bool) -> bool {
+        if demote_program_write_locks && self.is_invoked(index) {
+            return false;
+        }
```

Comment:
nit: not needed

Reply:
Ah, thanks! Yep, with https://github.com/solana-labs/solana/pull/19593/commits/db0f8d0afde705faf80743e891d5f31b1e517966 added this is redundant